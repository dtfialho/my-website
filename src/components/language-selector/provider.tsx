import { createContext, useState, ReactNode } from 'react'

type ContextProps = {
  showModal: boolean
  setShowModal: (value: boolean) => void
}

const LanguageSelectorContext = createContext<ContextProps>({
  showModal: false,
  setShowModal: () => {
    return
  }
})

type ProviderProps = {
  children: ReactNode
  initialState?: {
    showModal: boolean
  }
}

const LanguageSelectorProvider = ({
  children,
  initialState
}: ProviderProps) => {
  const [showModal, setShowModal] = useState(!!initialState?.showModal)

  return (
    <LanguageSelectorContext.Provider value={{ showModal, setShowModal }}>
      {children}
    </LanguageSelectorContext.Provider>
  )
}

export { LanguageSelectorProvider, LanguageSelectorContext }
