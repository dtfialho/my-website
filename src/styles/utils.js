// Colors
export const backgroundColor = '208,73,37'
export const backgroundColorDark = '153,46,36'
export const textColorLight = '255,255,255'
export const textColor = '100,100,100'
export const textColorDark = '51,51,51'

//Placeholders
export const container = `
  max-width: 1200px;
  margin: 0 auto;
  padding: 0 15px;
`

//Mixins
export const phoneOnly = (styles) => `
  @media (max-width: 599px) { ${styles} }
`
export const tabletPortrait = (styles) => `
  @media (min-width: 600px) { ${styles} }
`
export const tabletLandscape = (styles) => `
  @media (min-width: 900px) { ${styles} }
`
export const desktop = (styles) => `
  @media (min-width: 1200px) { ${styles} }
`
export const largeDesktop = (styles) => `
  @media (min-width: 1800px) { ${styles} }
`
