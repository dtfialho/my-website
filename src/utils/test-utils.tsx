import type { ReactNode } from 'react'
import { render, RenderResult } from '@testing-library/react'

const getFileTranslations = (locale = 'pt-BR', key: string) => {
  const [file, prop] = key.split(':')
  const data = require(`locales/${locale}/${file}.json`)
  return data[prop]
}

export const renderWithTranslate = (
  children: ReactNode,
  locale = 'pt-BR'
): RenderResult => {
  const useTranslation = jest.spyOn(
    require('next-translate/useTranslation'),
    'default'
  )
  useTranslation.mockImplementation(() => ({
    t: (key: string) => getFileTranslations(locale, key)
  }))

  return render(children)
}
