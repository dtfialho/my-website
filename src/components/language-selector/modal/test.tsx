import { screen, act } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { renderWithTranslate } from 'utils/test-utils'
import { LanguageSelectorProvider } from '../provider'
import Modal from './'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')
const defaultLocale = 'pt-BR'
const mockFn = jest.fn()

const mockRouter = (props?: any) =>
  useRouter.mockImplementation(() => ({
    locale: defaultLocale,
    locales: ['pt-BR', 'en'],
    pathname: '/test',
    asPath: '/test',
    query: {},
    push: mockFn,
    ...props
  }))

describe('Components/LanguageSelector/Modal', () => {
  it('Should render correctly', () => {
    mockRouter()

    const { container } = renderWithTranslate(
      <LanguageSelectorProvider initialState={{ showModal: true }}>
        <Modal />
      </LanguageSelectorProvider>
    )

    expect(container).toMatchSnapshot()
  })

  it('Should render correctly in en', () => {
    mockRouter({ locale: 'en' })

    const { container } = renderWithTranslate(
      <LanguageSelectorProvider initialState={{ showModal: true }}>
        <Modal />
      </LanguageSelectorProvider>,
      'en'
    )

    expect(container).toMatchSnapshot()
  })

  it('Should render with select language button disabled', () => {
    mockRouter()

    renderWithTranslate(
      <LanguageSelectorProvider initialState={{ showModal: true }}>
        <Modal />
      </LanguageSelectorProvider>
    )

    expect(
      screen.getByRole('button', { name: /Alterar idioma/ })
    ).toBeDisabled()
  })

  it('Should enable button on change selected language', async () => {
    mockRouter()

    renderWithTranslate(
      <LanguageSelectorProvider initialState={{ showModal: true }}>
        <Modal />
      </LanguageSelectorProvider>
    )

    await userEvent.click(screen.getByRole('button', { name: /Active locale/ }))
    await userEvent.click(await screen.findByAltText('en locale flag'))

    expect(
      screen.getByRole('button', { name: /Alterar idioma/ })
    ).not.toBeDisabled()
  })

  it('Should redirect to next location on change language and close modal modal', async () => {
    mockRouter()

    renderWithTranslate(
      <LanguageSelectorProvider initialState={{ showModal: true }}>
        <Modal />
      </LanguageSelectorProvider>
    )

    await userEvent.click(
      screen.getByRole('button', { name: /active locale/i })
    )
    await userEvent.click(await screen.findByAltText('en locale flag'))
    await userEvent.click(
      await screen.findByRole('button', { name: /alterar idioma/i })
    )

    expect(mockFn).toHaveBeenCalled()
    expect(
      screen.queryByText('Selecione a sua linguagem')
    ).not.toBeInTheDocument()
  })
})
