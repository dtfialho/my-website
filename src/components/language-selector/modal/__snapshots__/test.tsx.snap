// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/LanguageSelector/Modal Should render correctly 1`] = `
.c5 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  position: fixed;
  background-color: rgba(0,0,0,0.8);
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 2;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
}

.c1 {
  position: fixed;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 310px;
  background-color: #fff;
  z-index: 2;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
  border-radius: 10px;
  padding: 24px;
}

.c2 {
  width: 100%;
  position: relative;
  border-bottom: 1px solid #d9d9d9;
}

.c3 {
  font-size: 20px;
  margin-bottom: 8px;
}

.c4 {
  position: absolute;
  top: -16px;
  right: -16px;
  display: block;
  width: 25px;
  height: 25px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  border: none;
  cursor: pointer;
}

.c6 {
  padding-top: 16px;
}

.c7 {
  display: block;
  position: relative;
  width: 100%;
  margin-bottom: 16px;
}

.c8 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  border: 1px solid #d9d9d9;
  border-radius: 5px;
  padding: 8px;
}

.c9 {
  width: 29px;
  height: 22px;
  margin-right: 8px;
}

.c10 {
  display: block;
  position: absolute;
  top: 50%;
  right: 8px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c11 {
  position: absolute;
  top: 100%;
  left: 0;
  width: 100%;
  list-style-type: none;
  padding: 0;
  max-height: 0;
  background-color: #fff;
  overflow: hidden;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
}

.c11 li {
  opacity: 0;
  padding: 0 8px;
}

.c12 {
  font-size: 14px;
  padding: 8px;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  opacity: 1;
}

.c12:not(:last-child) {
  border-bottom: 1px solid #d9d9d9;
}

.c13 {
  font: 700 16px/16px 'Lato',sans-serif;
  text-align: center;
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  color: #fff;
  background-color: #992e24;
  border: none;
  border-radius: 5px;
  padding: 8px;
  cursor: pointer;
}

.c13:disabled {
  background: #fff;
  color: #ccc;
  border: 1px solid #d9d9d9;
  cursor: default;
}

<div>
  <div
    class="c0"
  />
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h2
        class="c3"
      >
        Selecione a sua linguagem
        :
      </h2>
      <button
        class="c4"
        type="button"
      >
        <svg
          class="c5"
          fill="none"
          focusable="false"
          height="25"
          role="img"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Close
          </title>
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </button>
    </div>
    <div
      class="c6"
    >
      <div
        class="c7"
      >
        <button
          class="c8"
          type="button"
        >
          <figure
            class="c9"
          >
            <img
              alt="Active locale"
              height="22"
              src="/img/pt-BR.jpg"
              title="Active locale"
              width="29"
            />
          </figure>
          pt-BR
          <span
            class="c10"
          >
            <svg
              aria-hidden="true"
              class="c5"
              fill="currentColor"
              focusable="false"
              height="15"
              viewBox="0 0 320 512"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                fill="currentColor"
              />
            </svg>
          </span>
        </button>
        <ul
          class="c11"
        >
          <li
            class="c12"
          >
            <figure
              class="c9"
            >
              <img
                height="22"
                src="/img/pt-BR.jpg"
                width="29"
              />
            </figure>
            pt-BR
          </li>
          <li
            class="c12"
          >
            <figure
              class="c9"
            >
              <img
                height="22"
                src="/img/en.jpg"
                width="29"
              />
            </figure>
            en
          </li>
        </ul>
      </div>
      <button
        class="c13"
        disabled=""
        type="button"
      >
        Alterar idioma
      </button>
    </div>
  </div>
</div>
`;

exports[`Components/LanguageSelector/Modal Should render correctly in en 1`] = `
.c5 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  position: fixed;
  background-color: rgba(0,0,0,0.8);
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 2;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
}

.c1 {
  position: fixed;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 310px;
  background-color: #fff;
  z-index: 2;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
  border-radius: 10px;
  padding: 24px;
}

.c2 {
  width: 100%;
  position: relative;
  border-bottom: 1px solid #d9d9d9;
}

.c3 {
  font-size: 20px;
  margin-bottom: 8px;
}

.c4 {
  position: absolute;
  top: -16px;
  right: -16px;
  display: block;
  width: 25px;
  height: 25px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  border: none;
  cursor: pointer;
}

.c6 {
  padding-top: 16px;
}

.c7 {
  display: block;
  position: relative;
  width: 100%;
  margin-bottom: 16px;
}

.c8 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  border: 1px solid #d9d9d9;
  border-radius: 5px;
  padding: 8px;
}

.c9 {
  width: 29px;
  height: 22px;
  margin-right: 8px;
}

.c10 {
  display: block;
  position: absolute;
  top: 50%;
  right: 8px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c11 {
  position: absolute;
  top: 100%;
  left: 0;
  width: 100%;
  list-style-type: none;
  padding: 0;
  max-height: 0;
  background-color: #fff;
  overflow: hidden;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
}

.c11 li {
  opacity: 0;
  padding: 0 8px;
}

.c12 {
  font-size: 14px;
  padding: 8px;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  opacity: 1;
}

.c12:not(:last-child) {
  border-bottom: 1px solid #d9d9d9;
}

.c13 {
  font: 700 16px/16px 'Lato',sans-serif;
  text-align: center;
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  color: #fff;
  background-color: #992e24;
  border: none;
  border-radius: 5px;
  padding: 8px;
  cursor: pointer;
}

.c13:disabled {
  background: #fff;
  color: #ccc;
  border: 1px solid #d9d9d9;
  cursor: default;
}

<div>
  <div
    class="c0"
  />
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h2
        class="c3"
      >
        Select your language
        :
      </h2>
      <button
        class="c4"
        type="button"
      >
        <svg
          class="c5"
          fill="none"
          focusable="false"
          height="25"
          role="img"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Close
          </title>
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </button>
    </div>
    <div
      class="c6"
    >
      <div
        class="c7"
      >
        <button
          class="c8"
          type="button"
        >
          <figure
            class="c9"
          >
            <img
              alt="Active locale"
              height="22"
              src="/img/en.jpg"
              title="Active locale"
              width="29"
            />
          </figure>
          en
          <span
            class="c10"
          >
            <svg
              aria-hidden="true"
              class="c5"
              fill="currentColor"
              focusable="false"
              height="15"
              viewBox="0 0 320 512"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                fill="currentColor"
              />
            </svg>
          </span>
        </button>
        <ul
          class="c11"
        >
          <li
            class="c12"
          >
            <figure
              class="c9"
            >
              <img
                height="22"
                src="/img/pt-BR.jpg"
                width="29"
              />
            </figure>
            pt-BR
          </li>
          <li
            class="c12"
          >
            <figure
              class="c9"
            >
              <img
                height="22"
                src="/img/en.jpg"
                width="29"
              />
            </figure>
            en
          </li>
        </ul>
      </div>
      <button
        class="c13"
        disabled=""
        type="button"
      >
        Change language
      </button>
    </div>
  </div>
</div>
`;
